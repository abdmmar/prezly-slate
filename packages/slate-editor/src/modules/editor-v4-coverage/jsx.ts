/* eslint-disable @typescript-eslint/no-namespace */

import { Coverage } from '@prezly/sdk';
import { createHyperscript } from '@prezly/slate-hyperscript';
import { COVERAGE_NODE_TYPE, PARAGRAPH_NODE_TYPE } from '@prezly/slate-types';
import { ReactNode } from 'react';

declare global {
    namespace JSX {
        interface IntrinsicElements {
            'h-coverage': {
                children?: ReactNode;
                coverage: {
                    id: Coverage['id'];
                };
                uuid: string;
            };
            'h-p': {
                children?: ReactNode;
            };
        }
    }
}

const jsx = createHyperscript({
    elements: {
        'h-coverage': { type: COVERAGE_NODE_TYPE },
        'h-p': { type: PARAGRAPH_NODE_TYPE },
    },
});

export default jsx;
