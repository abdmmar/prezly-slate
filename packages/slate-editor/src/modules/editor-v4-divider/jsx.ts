/* eslint-disable @typescript-eslint/no-namespace */

import { createHyperscript } from '@prezly/slate-hyperscript';
import { PARAGRAPH_NODE_TYPE } from '@prezly/slate-types';
import { ReactNode } from 'react';

import { DIVIDER_TYPE } from './constants';

declare global {
    namespace JSX {
        interface IntrinsicElements {
            'h-divider': {
                children?: ReactNode;
            };
            'h-p': {
                children?: ReactNode;
            };
        }
    }
}

const jsx = createHyperscript({
    elements: {
        'h-divider': { type: DIVIDER_TYPE },
        'h-p': { type: PARAGRAPH_NODE_TYPE },
    },
});

export default jsx;
