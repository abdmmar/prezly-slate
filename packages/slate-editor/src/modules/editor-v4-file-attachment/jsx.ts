/* eslint-disable @typescript-eslint/no-namespace */

import { createHyperscript } from '@prezly/slate-hyperscript';
import { ATTACHMENT_NODE_TYPE, PARAGRAPH_NODE_TYPE } from '@prezly/slate-types';
import type { UploadcareStoragePayload } from '@prezly/uploadcare';
import type { ReactNode } from 'react';

declare global {
    namespace JSX {
        interface IntrinsicElements {
            'h-file-attachment': {
                children?: ReactNode;
                description: string;
                file: UploadcareStoragePayload;
            };
            'h-p': {
                children?: ReactNode;
            };
        }
    }
}

const jsx = createHyperscript({
    elements: {
        'h-file-attachment': { type: ATTACHMENT_NODE_TYPE },
        'h-p': { type: PARAGRAPH_NODE_TYPE },
    },
});

export default jsx;
