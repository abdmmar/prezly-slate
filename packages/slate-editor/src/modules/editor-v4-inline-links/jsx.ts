/* eslint-disable @typescript-eslint/no-namespace,@typescript-eslint/no-unused-vars */
import { LINK_NODE_TYPE, PARAGRAPH_NODE_TYPE } from '@prezly/slate-types';
import { createEditor as makeEditor } from 'slate';
import { createHyperscript, createEditor } from 'slate-hyperscript';

import { InlineLinksExtension } from './InlineLinksExtension';

import { createEditorV4 } from '#modules/editor-v4/createEditorV4';

// This allows tests to include Slate Nodes written in JSX without TypeScript complaining.
declare global {
    namespace JSX {
        interface IntrinsicElements {
            [elemName: string]: any; // eslint-disable-line
        }
    }
}

const extensions = [InlineLinksExtension()];

export const jsx = createHyperscript({
    elements: {
        link: { type: LINK_NODE_TYPE },
        paragraph: { type: PARAGRAPH_NODE_TYPE },
    },
    creators: {
        editor: createEditor(() => createEditorV4(makeEditor(), () => extensions)),
    },
});
