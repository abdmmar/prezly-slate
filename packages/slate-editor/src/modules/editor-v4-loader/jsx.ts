/* eslint-disable @typescript-eslint/no-namespace */

import { createHyperscript } from '@prezly/slate-hyperscript';
import { PARAGRAPH_NODE_TYPE } from '@prezly/slate-types';
import { ReactNode } from 'react';

import { LOADER_TYPE } from './constants';
import { LoaderNode } from './types';

declare global {
    namespace JSX {
        interface IntrinsicElements {
            'h-loader': {
                children?: ReactNode;
                contentType: LoaderNode['contentType'];
                id: LoaderNode['id'];
                message: LoaderNode['message'];
            };
            'h-p': {
                children?: ReactNode;
            };
        }
    }
}

const jsx = createHyperscript({
    elements: {
        'h-loader': { type: LOADER_TYPE },
        'h-p': { type: PARAGRAPH_NODE_TYPE },
    },
});

export default jsx;
