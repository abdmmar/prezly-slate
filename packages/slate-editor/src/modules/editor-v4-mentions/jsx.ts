/* eslint-disable @typescript-eslint/no-namespace */

import { PARAGRAPH_TYPE } from '@prezly/slate-commons';
import { createHyperscript } from '@prezly/slate-hyperscript';
import { PlaceholderNode, PLACEHOLDER_NODE_TYPE } from '@prezly/slate-types';
import { ReactNode } from 'react';

declare global {
    namespace JSX {
        interface IntrinsicElements {
            'h-placeholder-mention': {
                children?: ReactNode;
                key: PlaceholderNode['key'];
            };
            'h-p': {
                children?: ReactNode;
            };
        }
    }
}

const jsx = createHyperscript({
    elements: {
        'h-placeholder-mention': { type: PLACEHOLDER_NODE_TYPE },
        'h-p': { type: PARAGRAPH_TYPE },
    },
});

export default jsx;
