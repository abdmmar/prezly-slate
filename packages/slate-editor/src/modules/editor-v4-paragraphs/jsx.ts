/* eslint-disable @typescript-eslint/no-namespace */

import { createHyperscript } from '@prezly/slate-hyperscript';
import { PARAGRAPH_NODE_TYPE } from '@prezly/slate-types';
import { ReactNode } from 'react';

declare global {
    namespace JSX {
        interface IntrinsicElements {
            'h-p': {
                children?: ReactNode;
            };
        }
    }
}

const jsx = createHyperscript({
    elements: {
        'h-p': { type: PARAGRAPH_NODE_TYPE },
    },
});

export default jsx;
