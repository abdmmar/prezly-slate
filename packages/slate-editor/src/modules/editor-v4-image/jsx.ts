/* eslint-disable @typescript-eslint/no-namespace */

import { PARAGRAPH_TYPE } from '@prezly/slate-commons';
import { createHyperscript } from '@prezly/slate-hyperscript';
import { UploadcareStoragePayload } from '@prezly/uploadcare';
import { ReactNode } from 'react';

import { LOADER_TYPE, LoaderElementType } from '../../modules/editor-v4-loader';

import { IMAGE_CANDIDATE_TYPE, IMAGE_TYPE } from './constants';
import { ImageLayout } from './types';

declare global {
    namespace JSX {
        interface IntrinsicElements {
            'h-image': {
                children?: ReactNode;
                file: UploadcareStoragePayload;
                href: string;
                layout: ImageLayout;
                width: string;
                width_factor: string;
            };
            'h-image-candidate': {
                children?: ReactNode;
                href: string;
                src: string;
            };
            'h-loader': {
                children?: ReactNode;
                contentType: LoaderElementType['contentType'];
                id: LoaderElementType['id'];
                message: LoaderElementType['message'];
            };
            'h-p': {
                children?: ReactNode;
            };
        }
    }
}

const jsx = createHyperscript({
    elements: {
        'h-image': { type: IMAGE_TYPE },
        'h-image-candidate': { type: IMAGE_CANDIDATE_TYPE },
        'h-loader': { type: LOADER_TYPE },
        'h-p': { type: PARAGRAPH_TYPE },
    },
});

export default jsx;
