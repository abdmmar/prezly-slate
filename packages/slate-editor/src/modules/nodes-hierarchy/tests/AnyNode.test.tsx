/* eslint-disable react/no-children-prop */
/** @jsx jsx */

import { Alignment } from '@prezly/slate-types';
import { Editor } from 'slate';

import { jsx } from '../../../jsx';

describe('nodes-hierarchy / Any Node', () => {
    it('should remove nodes without children', () => {
        const editor = (
            <editor>
                <h:ol align={Alignment.LEFT}>
                    <h:li children={[]} />
                    <h:li>
                        <h:li-text>
                            <h:text>Second</h:text>
                        </h:li-text>
                    </h:li>
                    <h:li>
                        <h:li-text>
                            <h:text>Third</h:text>
                        </h:li-text>
                    </h:li>
                </h:ol>
            </editor>
        ) as unknown as Editor;

        const expected = (
            <editor>
                <h:ol align={Alignment.LEFT}>
                    <h:li>
                        <h:li-text>
                            <h:text>Second</h:text>
                        </h:li-text>
                    </h:li>
                    <h:li>
                        <h:li-text>
                            <h:text>Third</h:text>
                        </h:li-text>
                    </h:li>
                </h:ol>
            </editor>
        ) as unknown as Editor;

        Editor.normalize(editor, { force: true });

        expect(editor.children).toMatchSnapshot();
        expect(editor.children).toEqual(expected.children);
        expect(editor.selection).toEqual(expected.selection);
    });

    it('should remove all nodes but leave editor with one paragraph', () => {
        const editor = (
            <editor>
                <h:ol align={Alignment.LEFT}>
                    <h:li>
                        <h:li-text>
                            <h:text children={[]} />
                        </h:li-text>
                    </h:li>
                    <h:li>
                        <h:li-text children={[]} />
                    </h:li>
                    <h:li children={[]} />
                </h:ol>

                <h:ul children={[]} />
                <h:paragraph children={[]} align={Alignment.CENTER} />
            </editor>
        ) as unknown as Editor;

        const expected = (
            <editor>
                <h:paragraph align={Alignment.CENTER} />
            </editor>
        ) as unknown as Editor;

        Editor.normalize(editor, { force: true });

        expect(editor.children).toMatchSnapshot();
        expect(editor.children).toEqual(expected.children);
        expect(editor.selection).toEqual(expected.selection);
    });
});
