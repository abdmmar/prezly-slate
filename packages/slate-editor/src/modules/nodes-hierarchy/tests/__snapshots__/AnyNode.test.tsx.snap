// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`nodes-hierarchy / Any Node should remove all nodes but leave editor with one paragraph 1`] = `
Array [
  Object {
    "align": "center",
    "children": Array [
      Object {
        "text": "",
      },
    ],
    "type": "paragraph",
  },
]
`;

exports[`nodes-hierarchy / Any Node should remove nodes without children 1`] = `
Array [
  Object {
    "align": "left",
    "children": Array [
      Object {
        "children": Array [
          Object {
            "children": Array [
              Object {
                "text": "Second",
              },
            ],
            "type": "list-item-text",
          },
        ],
        "type": "list-item",
      },
      Object {
        "children": Array [
          Object {
            "children": Array [
              Object {
                "text": "Third",
              },
            ],
            "type": "list-item-text",
          },
        ],
        "type": "list-item",
      },
    ],
    "type": "numbered-list",
  },
]
`;
